mapie/quantile_regression.py

```diff
@@ -0,0 +1,323 @@
+        ...
+        WARNING: Not correct.
+        """
+        if (
+            (y_pred_low >= y_pred_up).all() or
+                (y_preds[2] <= y_pred_low).all() or
+                (y_preds[2] >= y_pred_up).all()):
+            warnings.warn(
+                "WARNING: Not correct."
+            )
+
+    def _check_cv(
+        self,
+        cv: Optional[Union[int, str, BaseCrossValidator]] = None
+    ) -> Union[str, BaseCrossValidator]:
+        if cv is None:
+            cv == "split"
+            return cv
+        if cv == "split":
+            return cv
+        else:
+            raise ValueError(
+                "Invalid cv method."
+            )
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> MapieQuantileRegressor:
+        # Checks
+        self._check_parameters()
+        cv = self._check_cv(self.cv)
+        estimator = self._check_estimator(self.estimator)
+        alpha = self._check_alpha(self.alpha)
+        self._check_size_calib_set(y, self.alpha)
+        alpha_copy = alpha.copy()
+        alpha_copy.append(0.5)
+        # agg_function = self._check_agg_function(self.agg_function)
+        X, y = indexable(X, y)
+        y = _check_y(y)
+        self.n_features_in_ = check_n_features_in(X, cv, estimator)
+        sample_weight, X, y = check_null_weight(sample_weight, X, y)
+        y = cast(NDArray, y)
+        # n_samples = _num_samples(y)
+
+        # Initialization
+        self.estimators_: List[RegressorMixin] = []
+
+        # Work
+        if sample_weight is None:
+            (X_fit, X_calib,
+                y_fit, y_calib) = train_test_split(
+                X,
+                y,
+                test_size=0.5,
+                random_state=42
+            )
+            sample_weight_fit = sample_weight
+        else:
+            (X_fit, X_calib,
+                y_fit, y_calib,
+                sample_weight_fit, sample_weight_calib) = train_test_split(
+                X,
+                y,
+                sample_weight,
+                test_size=0.5,
+                random_state=42
+            )
+        list_estimators = []
+        list_y_preds_calib = []
+        name_estimator = estimator.__class__.__name__
+        for item in alpha_copy:
+            estimator_cloned = clone(estimator)
+            params = {
+                self.link_estimator_quantile[name_estimator][1]: item
+            }
+            estimator_cloned.set_params(**params)
+            estimator_cloned_ = fit_estimator(
+                estimator_cloned, X_fit, y_fit, sample_weight_fit
+            )
+            list_estimators.append(estimator_cloned_)
+            y_pred_calib = estimator_cloned_.predict(X_calib)
+            # check_nan_in_aposteriori_prediction(y_pred_calib)
+            list_y_preds_calib.append(y_pred_calib)
+            self.y_calib = y_calib
+
+        self.list_y_preds_calib = list_y_preds_calib
```

Comment:
Can be deleted as `self.list_y_preds_calib` is not used elsewhere (and is no longer needed by the user since the train/calib split is defined by the user)